src/Stratis.Bitcoin/Consensus/ConsensusManager.cs

```diff
@@ -756,7 +711,7 @@ private async Task<ConnectBlocksResult> ConnectBlockAsync(ChainedHeader chainTip
                 return failureResult;
             }
 
-            lock (this.peerLock)
+            lock (this.peerLock) //why do we lock here for instance ?
```

Comment:
Ask on slack, not in code, please.

And we need to lock here because all access to CHT should be protected by the `peerLock`.

Reply:
yes sure, I did planned to remove that